dragonfly/implementations/psyllid_provider.py

```diff
@@ -0,0 +1,279 @@
+                self.freq_dict[channel]=cf
+                return True
+
+            elif self.mode_dict[channel] == 'triggered':
+                request = 'active-config.ch'+self.channel_dict[channel]+'.trw.center-freq'
+                result = self.provider.set(self.queue_dict[channel]+'.'+request, cf)
+                logger.info('Set central frequency of egg writer for channel {} to {} Hz'.format(channel, cf))
+                self.freq_dict[channel]=cf
+                return True
+            else:
+                return False
+        except core.exceptions.DriplineError:
+            logger.error('Could not set central frequency of Psyllid instance for channel {}'.format(channel))
+            self.freq_dict[channel]=None
+            return False
+        except Exception as e:
+            logger.error('Unexpected error in set_central_frequency')
+            raise 
+
+
+    def start_run(self, channel, duration, filename):
+        payload = {'duration':duration, 'filename':filename}
+        result = self.provider.cmd(self.queue_dict[channel], 'start-run', payload=payload)
+
+
+    def stop_run(self, channel):
+        result = self.provider.cmd(self.queue_dict[channel], 'stop-run')
+
+
+    @property
+    def number_of_channels(self):
+        active_channels = [i for i in self.freq_dict.keys() if self.freq_dict[i]!=None]
+        logger.info('Active channels are {}'.format(active_channels))
+        return len(active_channels)
+
+
+    @property
+    def active_channels(self):
+        active_channels = [i for i in self.status_value_dict.keys() if self.status_value_dict[i]==4]
+        return active_channels
+
+
+    def get_number_of_streams(self, channel):
+        stream_count = 0
+        for i in range(3):
+            try:
+                request = '.node-config.ch'+str(i)+'.strw'
+                result = self.provider.get(self.queue_dict[channel]+request)
+                stream_count += 1
+                self.mode_dict[channel]='streaming'
+            except core.exceptions.DriplineError:
+                try:
+                    request = '.node-config.ch'+str(i)+'.trw'
+                    result = self.provider.get(self.queue_dict[channel]+request)
+                    stream_count += 1
+                    self.mode_dict[channel]='triggered'
+                except core.exceptions.DriplineError:
+                    pass  
+        logger.info('Number of streams for channel {}: {}'.format(channel, stream_count))
+        return stream_count
+
+
+    def set_fmt_snr_threshold(self, snr, channel='a'):
+        if self.mode_dict[channel] == 'streaming':
+            logger.warning('Psyllid not in streaming mode')
+            return False
+        request = '.node-config.ch'+str(self.channel_dict[channel])+'.fmt.threshold-power-snr'
+        result = self.provider.set(self.queue_dict[channel]+request, snr)
+        self.deactivate(channel)
```

Comment:
This should just be self.reactivate(channel). The last time I tested this I got into trouble with reactivating psyllid (psyllid/sepfile) and changed it to this. It was a quick test though and more testing of the psyllid mask writing and triggering is on my list. I intended to make it an issue then if the problem persists.

I could also remove theses methods for a later merge, as this one is not intended for usage for triggered acquisition.